operator/crd/crd.yaml

```diff
@@ -89,22 +89,37 @@ spec:
                       - claimName
                       - mountPath
                 tls:
-                  description: "Configure client TLS connections using a JKS or PKCS12 truststore. A JKS truststore should have its password set to 'changeit', while a PKCS12 truststore should have an empty password."
+                  description: "Configure server and client TLS connections."
                   type: object
                   properties:
-                    configMapName:
-                      type: string
-                    key:
-                      type: string
-                    type:
-                      type: string
-                      enum:
-                        - jks
-                        - pkcs12
-                  required:
-                    - configMapName
-                    - key
-                    - type
+                    clientConfig:
+                      description: "Configure client TLS connections using a JKS or PKCS12 truststore. A JKS truststore should have its password set to 'changeit', while a PKCS12 truststore should have an empty password."
+                      type: object
+                      properties:
+                        configMapName:
+                          type: string
+                        key:
+                          type: string
+                        type:
+                          type: string
+                          enum:
+                            - jks
+                            - pkcs12
+                      required:
+                        - configMapName
+                        - key
+                        - type
+                    serverConfig:
+                      description: "Configure server TLS connections using a JKS keystore. The keystore password should be stored in a secret and referenced in the route XML as a secretSource. The format of the secret should be `server.ssl.key-password=password`."
```

Comment:
:question: Would the secret be referenced in the route XML? Or in the IntegrationRoute CR?

Reply:
Good catch, it should be in the CR. I also see a typo in the example key.